src/pudl/metadata/classes.py

```diff
@@ -1632,3 +1655,39 @@ def to_sql(
                 check_values=check_values,
             )
         return metadata
+
+
+class CodeMetadata(Base):
+    """
+    A list of Encoders representing standardization and description for reported categorical codes.
+
+    Used to export to documentation.
+    """
+
+    encoder_list: List[Encoder] = []
+
+    @classmethod
+    def from_code_ids(
+        cls, code_ids: Iterable[str]
+    ) -> "CodeMetadata":
+        """
+        Construct a dictionary of code dictionaries containing the code name as it appears in the docs, a description, and file path to a CSV of the code label dataframe.
+
+        Args:
+            code_names: A list of Code PUDL identifiers, keys to entries in the CODE_METADATA dictionary.
```

Comment:
I think you mean `code_ids` here? Keeping with the same naming convention across all of these methods will help them be more readable.